src/main/scala/com/mongodb/spark/MongoSpark.scala

```diff
@@ -604,10 +607,7 @@ case class MongoSpark(sparkSession: SparkSession, connector: MongoConnector, rea
    * @return a DataFrame.
    */
   def toDF(schema: StructType): DataFrame = {
-    sparkSession.read.format("com.mongodb.spark.sql")
-      .schema(schema)
-      .options(readConfig.asOptions)
-      .load()
+    sparkSession.baseRelationToDataFrame(MongoRelation(toBsonDocumentRDD, Option(schema))(sparkSession.sqlContext).asInstanceOf[BaseRelation])
```

Comment:
This essentially shortcuts the Spark logic for getting the dataframe from a relation.

Reply:
Ack